### PR TITLE
New rspec integration form

### PR DESCRIPTION
**Pull request by coreyhaines**
Extract RSpec integration from the objects that know what to display. This isolates the display characters into an object that can be reused moving forward to minitest support.
This also introduces a helper method for building rspec formatters easily.

### CODE REVIEW

**Comment by coreyhaines** [comment]
This is an interim pull-request. While the code is working, we haven't moved the gem over to using this stuff, yet. I'd like to get this integrated before a rename and removal of the old code. Then, we can move forward to minitest support.
There is also a problem with the name-inference if the display provider class is namespaced. There is a pending spec for it, and I'll work on adding the code for it.

File: .rspec [hunk 1]
```diff
@@ -1,2 +1,3 @@
 --color
 --order rand
+-r emoji-rspec
```
Comment (cupakromer) [inline .rspec:3]: Was this added for the specs or for dog fooding?

**Comment by cupakromer** [comment]
Looks good. I've got some stuff in the works. I'll merge this and rebase mystuff on top.

File: .rspec (hunk 1 above)
Comment (coreyhaines) [inline .rspec:3]: Yeah, for dog-fooding. Makes it easier to run the example spec file

Comment (coreyhaines) [inline .rspec:3]: Should change it to include the emoji-test-love

**Pull request merged**